File: pset6/mario/mario.py
# Pset6: Mario More
from cs50 import get_int


def main():
    height = -1

    # Ask the height
    while ((height < 1) or (height > 8)):
        height = get_int("Height : ")

    for i in range(0, height):
        for col in range(0,  ((height * 2) + 2)):
            # if is the first column
            if (col < height):
                if (col < height - i - 1):
                    print(" ", end='')
                else:
                    print("#", end='')
            # if is the second column
            elif (col == height):
                print("  ", end='')
            else:
                if (col < height + 2 + i):
                    print("#", end='')
        # Go to next line
        print("")


if __name__ == "__main__":
    main()

File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int height;

    // Ask the height
    do
    {
        height = get_int("Height : ");
    }
    while ((height < 1) || (height > 8));



    // build line
    for (int i = 0; i < height; i++)
    {
        for (int col = 0; col < ((height * 2) + 2); col++)
        {

            // if is the first column
            if (col < height)
            {
                if (col < height - i - 1)
                {
                    printf(" ");
                }
                else
                {
                    printf("#");
                }
            }
            // if is the second column
            else if (col == height)
            {
                printf("  ");
            }
            else
            {
                if (col < height + 2 + i)
                {
                    printf("#");
                }
            }
        }

        // Go to next line
        printf("\n");
    }
}